packages/flutter/lib/src/rendering/box.dart

```diff
@@ -1970,16 +1970,26 @@ abstract class RenderBox extends RenderObject {
         assert(size._owner == this);
         if (RenderObject.debugActiveLayout != null &&
             !RenderObject.debugActiveLayout!.debugDoingThisLayoutWithCallback) {
-          assert(
-            debugDoingThisResize || debugDoingThisLayout || _computingThisDryLayout ||
-              (RenderObject.debugActiveLayout == parent && size._canBeUsedByParent),
-            'RenderBox.size accessed beyond the scope of resize, layout, or '
-            'permitted parent access. RenderBox can always access its own size, '
-            'otherwise, the only object that is allowed to read RenderBox.size '
-            'is its parent, if they have said they will. It you hit this assert '
-            'trying to access a child\'s size, pass "parentUsesSize: true" to '
-            "that child's layout().",
-          );
+          // The Flutter pipeline is currently running a layout pass. 
+          // TODO: explain why we care that debugDoingThisLayoutWithCallback
```

Comment:
Tracked it back to https://github.com/flutter/flutter/pull/74402 for full context. Some widgets like RenderEditableTextState will access RenderBox.size when rebuilding.

Reply:
So that property refers to a "rebuild"? Can you define that term for me? I've seen it used a few different ways.

Reply:
I cannot, I am not familiar with the specific change. The docs for `debugDoingThisLayoutWithCallback` say `Whether invokeLayoutCallback for this render object is currently running.` Certainly room to improve the docs there though. I'll file a bug.

Reply:
`invokeLayoutCallback` is quite a bit more diverse than just the concept of "rebuild", I believe. 

`invokeLayoutCallback` may run by a `RenderObject` any number of times within that `RenderObject`'s `layout()` behavior. Inside of that callback, you're permitted to mutate the element and render tree during layout, which is otherwise prohibited.

The double and triple negatives are throwing me, so I still can't tell the intention. It looks like if we're in a layout callback, we don't bother checking any conditions? We let the `size` access go through no matter what? This is the kind of stuff I was trying to improve. The intention of these conditions is really hard to figure out.

Reply:
Filed https://github.com/flutter/flutter/issues/128913